src/WebJobs.Script/Workers/Http/DefaultHttpWorkerService.cs

```diff
@@ -170,5 +172,40 @@ private void AddRequestHeadersAndSetRequestUri(HttpRequestMessage httpRequestMes
             httpRequestMessage.Headers.Add(HttpWorkerConstants.HostVersionHeaderName, ScriptHost.Version);
             httpRequestMessage.Headers.UserAgent.ParseAdd($"{HttpWorkerConstants.UserAgentHeaderValue}/{ScriptHost.Version}");
         }
+
+        public async Task<bool> IsWorkerReady()
+        {
+            await Utility.DelayAsync(WorkerConstants.WorkerInitTimeoutSeconds, WorkerConstants.PollingIntervalMilliseconds, async () =>
+            {
+                return await ShouldContinueWaitingForWorker();
```

Comment:
What will happen in the case of a host restart while this is looping? This service is not being signaled, and this will keep looping until it times out. We need to signal the disposal/shutdown here so we can short circuit this process.

Reply:
thanks! Updated to pass CancellationToken from ScriptHost